File: scripts/phase_shift_figure.R
# =============================================================================
# JetLagPro — Phase-shift schematic (circadian misalignment + non-photic cues)
# =============================================================================
# Conceptual figure for the paper: origin circadian phase, destination
# misalignment, and timed stimulation windows as "Scheduled non-photic cues
# (acupressure)" so chronobiology reviewers see the protocol as a behavioral
# zeitgeber schedule.
#
# REQUIREMENTS: tidyverse (ggplot2)
#
# USAGE: Rscript scripts/phase_shift_figure.R
# Output: figures/phase_shift_schematic.png
# =============================================================================

library(tidyverse)

# Destination local time (0–24 h)
hour <- seq(0, 24, by = 0.1)

# Example: +6 h eastward. Origin phase lags destination clock by 6 h.
# So at destination hour 6, body is still at origin midnight.
shift_h <- 6
origin_phase    <- sin(((hour + shift_h) / 24) * 2 * pi)   # body still on origin clock
dest_phase      <- sin((hour / 24) * 2 * pi)               # target (destination-aligned)

df <- tibble(
  hour = rep(hour, 2),
  value = c(origin_phase, dest_phase),
  curve = rep(c("Origin circadian phase", "Destination (target) phase"), each = length(hour))
)

# Stimulation windows: 2 h each, meridian order LU, LI, ST, SP, HT, SI
# (example slots in destination local time; adjust to match Table 1 if needed)
windows <- tibble(
  xmin = c(2, 4, 6, 8, 10, 12),
  xmax = c(4, 6, 8, 10, 12, 14),
  meridian = c("LU", "LI", "ST", "SP", "HT", "SI")
)

fig_dir <- "figures"
if (!dir.exists(fig_dir)) dir.create(fig_dir, recursive = TRUE)

p <- ggplot(df, aes(x = hour, y = value, colour = curve, linetype = curve)) +
  geom_line(linewidth = 1) +
  scale_linetype_manual(values = c("Origin circadian phase" = "solid", "Destination (target) phase" = "dashed")) +
  scale_colour_manual(values = c("Origin circadian phase" = "black", "Destination (target) phase" = "gray40")) +
  geom_rect(
    data = windows,
    aes(xmin = xmin, xmax = xmax, ymin = -Inf, ymax = Inf),
    inherit.aes = FALSE,
    fill = "steelblue",
    alpha = 0.2
  ) +
  geom_text(
    data = windows,
    aes(x = (xmin + xmax) / 2, y = -1.15, label = meridian),
    inherit.aes = FALSE,
    size = 3,
    fontface = "bold"
  ) +
  annotate("text", x = 8, y = -1.35, label = "Scheduled non-photic cues (acupressure)", size = 3.2, fontface = "italic") +
  labs(
    x = "Local time (destination, h)",
    y = "Circadian phase (relative)",
    title = "Circadian misalignment after eastward travel (+6 h)",
    colour = NULL,
    linetype = NULL
  ) +
  scale_x_continuous(breaks = seq(0, 24, 4), limits = c(0, 24)) +
  scale_y_continuous(limits = c(-1.5, 1.2)) +
  theme_minimal(base_size = 11) +
  theme(
    legend.position = "bottom",
    plot.title = element_text(hjust = 0.5, size = 12)
  )

out_path <- file.path(fig_dir, "phase_shift_schematic.png")
ggsave(out_path, p, width = 7, height = 5, dpi = 150)
message("Saved ", out_path)

# Copy to assets/images/ for the paper webpage if that directory exists
assets_dir <- "assets/images"
if (dir.exists(assets_dir)) {
  file.copy(out_path, file.path(assets_dir, "phase_shift_schematic.png"), overwrite = TRUE)
  message("Copied to ", file.path(assets_dir, "phase_shift_schematic.png"))
}

File: scripts/phase_shift_figure.py
"""
JetLagPro — Phase-shift schematic (circadian misalignment + non-photic cues).
Generates the same figure as phase_shift_figure.R for the paper (Figure 2).
Output: figures/phase_shift_schematic.png (and copy to assets/images/ if present).
"""
import math
import os

try:
    import matplotlib
    matplotlib.use("Agg")
    import matplotlib.pyplot as plt
    import numpy as np
except ImportError:
    print("Install matplotlib and numpy: pip install matplotlib numpy")
    raise

# Destination local time (0–24 h)
hour = np.arange(0, 24.01, 0.1)
shift_h = 6  # eastward +6 h example
origin_phase = np.sin(((hour + shift_h) / 24) * 2 * math.pi)
dest_phase = np.sin((hour / 24) * 2 * math.pi)

fig, ax = plt.subplots(figsize=(7, 5))
ax.plot(hour, origin_phase, "k-", linewidth=1.5, label="Origin circadian phase")
ax.plot(hour, dest_phase, color="gray", linestyle="--", linewidth=1.5, label="Destination (target) phase")

# Stimulation windows: 2 h each, LU, LI, ST, SP, HT, SI
windows = [(2, 4), (4, 6), (6, 8), (8, 10), (10, 12), (12, 14)]
labels = ["LU", "LI", "ST", "SP", "HT", "SI"]
for (xmin, xmax), lab in zip(windows, labels):
    ax.axvspan(xmin, xmax, alpha=0.2, color="steelblue")
    ax.text((xmin + xmax) / 2, -1.15, lab, ha="center", fontsize=10, fontweight="bold")
ax.text(8, -1.35, "Scheduled non-photic cues (acupressure)", ha="center", fontsize=10, style="italic")

ax.set_xlabel("Local time (destination, h)")
ax.set_ylabel("Circadian phase (relative)")
ax.set_title("Circadian misalignment after eastward travel (+6 h)")
ax.set_xticks(range(0, 25, 4))
ax.set_ylim(-1.5, 1.2)
ax.legend(loc="lower right")
ax.grid(True, alpha=0.3)

fig_dir = "figures"
os.makedirs(fig_dir, exist_ok=True)
out_path = os.path.join(fig_dir, "phase_shift_schematic.png")
fig.savefig(out_path, dpi=150, bbox_inches="tight")
plt.close()
print("Saved", out_path)

assets_dir = "assets/images"
if os.path.isdir(assets_dir):
    import shutil
    dest = os.path.join(assets_dir, "phase_shift_schematic.png")
    shutil.copy2(out_path, dest)
    print("Copied to", dest)
